cmd/auth-provider-gcp/provider/provider.go

```diff
@@ -96,5 +96,6 @@ func GetResponse(image string, provider credentialconfig.DockerConfigProvider) (
 	}
 	response.TypeMeta.Kind = apiKind
 	response.TypeMeta.APIVersion = apiVersion
+	response.CacheKeyType = credentialproviderapi.RegistryPluginCacheKeyType
```

Comment:
https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/kubelet/pkg/apis/credentialprovider/types.go#L39

Is hardcoding this to registry the best option. I buy it may be the right default but I would imagine we would like this to either be command line, environment or file configurable. Not sure Registry is always the right answer out of Image/Registry/Global. As an example Registry ignores path and project is part of the path. So I'm not sure we won't hit cases where we want image.

Reply:
Not hardcoding `CacheKeyType` sounds like a good idea to me, though I'm not sure what mechanism we could use to change it. I was thinking about an environment variable read by `kube-up` or something, though would that be too cumbersome? We could only change `CacheKeyType` that way by restarting the cluster, unfortunately.